CONTRIBUTING.md

```diff
@@ -1,78 +1,122 @@
+
+### Documentation & Miscellaneous
+ - If you have suggestions for improvements to the documentation, tutorial or examples (or something
+   else), we would love to hear about it.
+ - As always first file a [Github issue].
+ - Implement the changes to the documentation, please read the [How To](#how-to-contribute).
+
+## How To Contribute
+
+### Requirements
 For a contribution to be accepted:
 
- - Take note of the [Development Guidelines](#development-guidelines)
- - Code must follow existing styling conventions
- - Commit message should start with a [issue number](#how-to) and should also be descriptive.
+- Documentation should always be updated or added.*
+- Examples should always be updated or added.*
+- Tests should always be updated or added.*
+- Format the Dart code accordingly with `flutter format`.
+- Your code should pass the analyzer checks `flutter analyze`.
+- Start your PR title with a [conventional commit] type
+  (`feat:`, `fix:` etc).
 
-If the contribution doesn't meet these criteria, a maintainer will discuss it with you on the issue.
-You can still continue to add more commits to the branch you have sent the Pull Request from.
+*When applicable.
 
-# How To Contribute
- - First of all [file an bug or feature report](https://github.com/flame-engine/flame/issues/new) on
- this repository.
- - [Fork the project](https://guides.github.com/activities/forking/#fork) on Github
+If the contribution doesn't meet these criteria, a maintainer will discuss it with you on the issue
+or PR. You can still continue to add more commits to the branch you have sent the Pull Request from
+and it will be automatically reflected in the PR.
+
+## Open an issue and fork the repository
+ - If it is a bigger change or a new feature, first of all
+   [file a bug or feature report][GitHub issues]
+ - [Fork the project][fork guide] on GitHub.
  - Clone the forked repository to your local development machine
- (e.g. `git clone https://github.com/<YOUR_GITHUB_USER>/flame.git`)
- - Run `flutter pub get` in the cloned repository to get all the dependencies
- - Create a new local branch based on issue number from first step
- (e.g. `git checkout -b 12-new-feature`)
- - Make your changes
- - When committing your changes, make sure to start the commit message with `#<issue-number>`
- (e.g. `git commit -m '#12 - New Feature added'`)
+   (e.g. `git clone git@github.com:<YOUR_GITHUB_USER>/flame.git`).
+
+### Environment Setup
+Flame uses [Melos] to manage the project and dependencies.
+
+To install Melos, run the following command from your terminal:
+
+```bash
+flutter pub global activate melos
+```
+
+Next, at the root of your locally cloned repository bootstrap the projects dependencies:
+
+```bash
+melos bootstrap
+```
+
+The bootstrap command locally links all dependencies within the project without having to
+provide manual [`dependency_overrides`][pubspec doc]. This allows all
+plugins, examples and tests to build from the local clone project.
```

Comment:
we should probably mention here, because it is unclear to me, when do we need to run this command other than once (or if only once). also does it create changes to our pubspec files that we would not want to commit?